commit 3b2f0a1cf5c59fb4bca311fd3088b46b7d599f93 (refs/changes/25/1725/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-03-27T21:40:12+02:00 (2 years, 6 months ago)
    
    TOOLS-1731 sdcadm post-setup common-external-nics should set external network as primary

diff --git a/lib/cli/do_update_other.js b/lib/cli/do_update_other.js
index 7e33b65..6096003 100644
--- a/lib/cli/do_update_other.js
+++ b/lib/cli/do_update_other.js
@@ -752,6 +752,43 @@ function do_update_other(subcmd, opts, args, cb) {
             }, next);
         },
 
+        function updateExternalNics(ctx, next) {
+            var nicSvcs = ctx.svcs.filter(function (svc) {
+                return (svc.name === 'adminui' || svc.name === 'imgapi');
+            });
+
+            var needUpdate = nicSvcs.some(function (svc) {
+                return (svc.params.networks.length === 1 ||
+                        typeof (svc.params.networks[0]) === 'string' ||
+                        typeof (svc.params.networks[1]) === 'string');
+            });
+
+            if (!needUpdate) {
+                next();
+                return;
+            }
+
+            progress(
+                'Fix networks parameter for "imgapi" and "adminui" services.');
+            vasync.forEachPipeline({
+                inputs: nicSvcs,
+                func: function updateSvcNics(svc, nextSvc) {
+                    updateService(svc.uuid, { params: {
+                        networks: [
+                            { name: 'admin' },
+                            { name: 'external', primary: true}
+                        ]
+                    } }, function (svcErr) {
+                        if (svcErr) {
+                            nextSvc(new errors.SDCClientError(svcErr, 'sapi'));
+                            return;
+                        }
+                        nextSvc();
+                    });
+                }
+            }, next);
+        },
+
         steps.agentServicesEnsureCreated
 
     ]}, function (err) {
diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 46e87c8..e8cf606 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -6162,7 +6162,10 @@ setupCommonExternalNics(opts, cb) {
 
             self.sapi.updateService(svcArr[0].uuid, {
                 params: {
-                    networks: ['admin', 'external']
+                    networks: [
+                        { name: 'admin' },
+                        { name: 'external', primary: true}
+                    ]
                 }
             }, function (err) {
                 if (err) {
